account_move_line_tax_editable/models/account_move_line.py

```diff
@@ -12,7 +12,63 @@ class AccountMoveLine(models.Model):
         string="Is tax data editable?", compute="_compute_is_tax_editable"
     )
 
+    tax_line_id = fields.Many2one(inverse="_inverse_tax_line_id")
+
     @api.depends("move_id.state")
     def _compute_is_tax_editable(self):
         for rec in self:
             rec.is_tax_editable = rec.move_id.state == "draft"
+
+    def _inverse_tax_line_id(self):
+        for rec in self:
+            repartition_type = rec.tax_repartition_line_id.repartition_type or "tax"
+            factor_percent = rec.tax_repartition_line_id.factor_percent or 100
+            has_account = bool(rec.tax_repartition_line_id.account_id)
+            if rec.move_id.move_type in ("out_refund", "in_refund"):
+                repartition_lines = (
+                    rec.tax_line_id.refund_repartition_line_ids.filtered(
+                        lambda rl: rl.repartition_type == repartition_type
+                        and rl.factor_percent == factor_percent
+                    )
+                )
+                if len(repartition_lines) > 1:
+                    rec.tax_repartition_line_id = (
+                        repartition_lines.filtered(
+                            lambda rl: rl.repartition_type == "base"
+                            or has_account
+                            is bool(rec.tax_repartition_line_id.account_id)
+                        )[:1]
+                        or repartition_lines[:1]
+                    )
+                elif repartition_lines:
+                    rec.tax_repartition_line_id = repartition_lines
+                else:
+                    rec.tax_repartition_line_id = (
+                        rec.tax_line_id.refund_repartition_line_ids.filtered(
+                            lambda rl: rl.repartition_type == repartition_type
+                        )[:1]
+                    )
```

Comment:
This part seems to be repeated 2 times. Can you do a more readable code avoiding duplicated parts?